[clue_nrf52840] Integrate tock bootloader

alexandruradovici opened the pull request:
### Pull Request Overview

This pull request adds `tock-bootloader` for the Adafruit Clue nRF52840 board.

The board was using the Adafruit nRF52 bootloader to flash the kernel and the apps. Now this bootloader is used to flash the `tock-bootloader`. I decided to keep the original bootloader as users of this board might want to switch to CircuitPython and back. 

The original bootloader allows flashing using copy and paste and keeps the board compatible with Adafruit's tutorials.

Even tough I am not a fan of shipping binary files, building the `tock-bootloader` for this board requires and older version of the kernel and can be somehow difficult. This is why this uses the same idea from the Nano 33 BLE board  and ships a binary files (bin and uf2).

### Testing Strategy

This pull request was tested using an Adafruit Clue nRF52480 board


### TODO or Help Wanted

This pull request still needs #2774 and #2776.


### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

lschuermann commented:
> Even tough I am not a fan of shipping binary files, building the `tock-bootloader` for this board requires and older version of the kernel and can be somehow difficult. This is why this uses the same idea from the Nano 33 BLE board and ships a binary files (bin and uf2).

Why can't we use the approach of the MicroBit of #2376? I think that turned out pretty great.

alexandruradovici commented:
> > Even tough I am not a fan of shipping binary files, building the `tock-bootloader` for this board requires and older version of the kernel and can be somehow difficult. This is why this uses the same idea from the Nano 33 BLE board and ships a binary files (bin and uf2).
> 
> Why can't we use the approach of the MicroBit of #2376? I think that turned out pretty great.

I can set up a workflow for that.

bradjc commented:
Should this be in 2.0?

alexandruradovici commented:
> Should this be in 2.0?

It much easier for users to use the board. I ran all the tests using this bootloader.

alexandruradovici commented:
I am trying to fix the cargo dependencies for the bootloader, but I am not able to find the [remove-submodule](https://github.com/tock/tock/commit/44fd8c3abb9b6626c6956b206d284bbdbd0bdb5d) branch for `tock` anymore. Was this branch removed?

hudson-ayers commented:
bors removes branches once they are merged. The changes in that branch were merged as part of this PR: https://github.com/tock/tock/pull/2323 . You should be able to just target that merge commit.

alexandruradovici commented:
I added a fixed Tock version and a workflow in https://github.com/tock/tock-bootloader/pull/25. As soon as it gets merged and someone runs the Release action, I will change this pull request to download the bootloader.

bradjc commented:
Now that we have https://github.com/tock/tock-bootloader/releases/tag/clue_nrf52840-1.1.2 can you double check that it works?

alexandruradovici commented:
> Now that we have https://github.com/tock/tock-bootloader/releases/tag/clue_nrf52840-1.1.2 can you double check that it works?

It works. I rebased it and squashed the commits so that the binary bootloader is not stored anywhere.

bradjc commented:
bors r+
